### PR TITLE
feat: endpoint implementation to check user attendance

### PR DESCRIPTION
**Pull request by anversa-pro**
## Describe your changes
Creation of a new API endpoint to handle user registration verification integrating event ID parameter
Implementation of validation logic to check user registration status and return result based on that status

## Issue ticket number and link
https://trello.com/c/ZiAjqgSk

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -31,18 +33,23 @@ app.use(async (request, response, next) => {
...
-  const token = request.headers['authorization']
+  try {
+    const token = request.headers['authorization'].replace('Bearer ', '')
 
-  if (!token) {
-    response.status(401).json({ status: '401', message: 'Unauthorized', data: {} })
-    return
-  }
+    const authService = await serviceContainer('authentication')
+    const authVerifyResponse = await authService.verifyToken(token)
```
Comment (jlora-mck) [inline app.js:40]: I dont get this logic here, in line 37 you are cleaning the token, then here you verify it ?

File: controllers/v1/users/user.controller.js
Comment (jlora-mck) [inline controllers/v1/users/user.controller.js:1]: why we are not using typescript here ?

File: services/user.service.js [hunk 2]
```diff
@@ -37,20 +37,20 @@ class UserService extends BaseService {
...
 
-      userRefSnapshot.forEach((doc) => {
-        let userData = doc.data();
-        userData.id = doc.id;
-        allUsers.push(userData);
-      });
+      userRefSnapshot.forEach(doc => {
+        let userData = doc.data()
+        userData.id = doc.id
+        allUsers.push(userData)
```
Comment (jlora-mck) [inline services/user.service.js:49]: i think we can refactor this to this 
```suggestion
        allUsers.push(doc.data())
```

File: services/user.service.js [hunk 3]
```diff
@@ -59,45 +59,70 @@ class UserService extends BaseService {
...
-      const successMessage = 'Getting user information successfully';
-      response = this.getSuccessResponse(user, successMessage);
+      const successMessage = 'Getting user information successfully'
+      response = this.getSuccessResponse(user, successMessage)
     } catch (err) {
-      const errorMessage = 'Error getting user information';
+      const errorMessage = 'Error getting user information'
       /* eslint-disable no-console */
       // console.error(errorMessage, err);
       /* eslint-enable */
```
Comment (jlora-mck) [inline services/user.service.js:85]: if not using lets remove

File: services/user.service.js [hunk 4]
```diff
@@ -59,45 +59,70 @@ class UserService extends BaseService {
...
 
     return response
   }
+  async checkUserAttendeeStatus(eventId, userId, eventsService) {
+    let response
+
+    try {
+      const eventData = await eventsService.findById(eventId)
+      if (!eventData) {
+        return this.getSuccessResponse(
```
Comment (jlora-mck) [inline services/user.service.js:97]: I think the naming can change a bit here, we are firing an error with a success response ?
change this name to reflect better what is this doing `getSuccessResponse `

File: services/user.service.js [hunk 5]
```diff
@@ -59,45 +59,70 @@ class UserService extends BaseService {
...
+          { attendanceConfirmed: false },
+          'No existing data for this event'
+        )
+      }
+
+      const userData = eventData.data.attendees.find(attendee => attendee === userId)
```
Comment (jlora-mck) [inline services/user.service.js:103]: add checks for when data is null
```suggestion
      const userData = eventData.data?.attendees?.find(attendee => attendee === userId)
```

File: services/user.service.js [hunk 6]
```diff
@@ -59,45 +59,70 @@ class UserService extends BaseService {
...
   async findByUserId(userId) {
     let user = null
     let response
 
     try {
-      let userRefSnapshot = await this.collection
-        .where('uid', '==', userId)
-        .limit(1)
-        .get()
+      let userRefSnapshot = await this.collection.where('uid', '==', userId).limit(1).get()
```
Comment (jlora-mck) [inline services/user.service.js:125]: why a let? i dont see that you are changing userRefSnapshot value elsewhere

Comment (jlora-mck) [reply services/user.service.js:125]: Also can we perform assertions with ===

File: services/user.service.js [hunk 7]
```diff
@@ -132,27 +157,26 @@ class UserService extends BaseService {
   async toggleEnable(id) {
     let response
     try {
-      const userInfoRef = await this.collection.doc(id).get();
+      const userInfoRef = await this.collection.doc(id).get()
 
-      const userData = userInfoRef.data();
-      userData.isEnabled = !userData.isEnabled;
+      const userData = userInfoRef.data()
+      userData.isEnabled = !userData.isEnabled
```
Comment (jlora-mck) [inline services/user.service.js:163]:
```suggestion
      const { isEnabled } = userInfoRef.data()
```

File: services/user.service.js [hunk 8]
```diff
@@ -132,27 +157,26 @@ class UserService extends BaseService {
...
 
       await this.collection.doc(id).update({
-        isEnabled: userData.isEnabled
-      });
-
-      const successMessage = 'User was disabled successfully';
-      response = this.getSuccessResponse(userData, successMessage);
+        isEnabled: userData.isEnabled,
```
Comment (jlora-mck) [inline services/user.service.js:166]:
```suggestion
        isEnabled,
```

Comment (jlora-mck) [reply services/user.service.js:166]: Taking into account changed i suggested before this

File: controllers/v1/users/user.controller.js
Comment (jose-sal-y-rosas) [reply controllers/v1/users/user.controller.js:1]: No. It is JS

Comment (jlora-mck) [reply controllers/v1/users/user.controller.js:1]: Yeah but why we are not using ts files ?

File: app.js (hunk 1 above)
Comment (anversa-pro) [reply app.js:40]: I'm removing the 'Bearer' text at the beginning, and then passing the token to verification

File: services/user.service.js (hunk 2 above)
Comment (anversa-pro) [reply services/user.service.js:49]: Jose, all this is based on firebase connection, all should be refactored for mongo indeed. including your suggestion, but is not the current Sprint goal, can we solve this and all changes not related to the event user and event validation as a refactor in backlog?

File: app.js (hunk 1 above)
Comment (jlora-mck) [reply app.js:40]: Ahhh ok sry i definitely misunderstood the flow 👍

File: services/user.service.js (hunk 4 above)
Comment (anversa-pro) [reply services/user.service.js:97]: I think is not an error, the process worked find but couldn't find a match, should that be considered and error?

File: services/user.service.js (hunk 2 above)
Comment (jlora-mck) [reply services/user.service.js:49]: Thats ok, but actually making this change in particular wont take you a lot of time from my perspective, or will it ?

File: services/user.service.js (hunk 4 above)
Comment (jlora-mck) [reply services/user.service.js:97]: is this returning as a 200 code ?

Comment (anversa-pro) [reply services/user.service.js:97]: yes!

File: services/user.service.js (hunk 2 above)
Comment (anversa-pro) [reply services/user.service.js:49]: Is not about time, is about the scope, but I can do it.

File: services/user.service.js (hunk 6 above)
Comment (anversa-pro) [reply services/user.service.js:125]: Not in scope, should be refactored in another ticket in backlog

File: services/user.service.js (hunk 7 above)
Comment (anversa-pro) [reply services/user.service.js:163]: Not in scope, should be refactored in another ticket in backlog

File: services/user.service.js (hunk 8 above)
Comment (anversa-pro) [reply services/user.service.js:166]: Not in scope, should be refactored in another ticket in backlog

File: services/user.service.js (hunk 4 above)
Comment (jlora-mck) [reply services/user.service.js:97]: mmm ok got it, as a frontend i would have loved to received another httpcode, someething that fails and goes into the catch of the try catch, but this is really up to you guys, if this is what you agreed thats ok

**Review by jlora-mck — APPROVED** [review]

**Review by FelipeLosada1996 — APPROVED** [review]

File: services/user.service.js (hunk 5 above)
Comment (anversa-pro) [reply services/user.service.js:94]: Implementation pending to make it work

**Pull request merged**